server/src/domain/system-config/system-config.core.ts

```diff
@@ -341,19 +342,19 @@ export class SystemConfigCore {
     if (force || !this.configCache) {
       try {
         const file = await this.repository.readFile(filepath);
-        const json = JSON.parse(file.toString());
+        const config = loadYaml(file.toString()) as any;
         const overrides: SystemConfigEntity<SystemConfigValue>[] = [];
 
         for (const key of Object.values(SystemConfigKey)) {
-          const value = _.get(json, key);
-          this.unsetDeep(json, key);
+          const value = _.get(config, key);
+          this.unsetDeep(config, key);
           if (value !== undefined) {
             overrides.push({ key, value });
           }
         }
 
-        if (!_.isEmpty(json)) {
-          this.logger.warn(`Unknown keys found: ${JSON.stringify(json, null, 2)}`);
+        if (!_.isEmpty(config)) {
+          this.logger.warn(`Unknown keys found: ${JSON.stringify(config, null, 2)}`);
```

Comment:
I left this as just JSON output rather than trying to format based on what the input was, to keep the code simple.

Reply:
I think that's fine, it is just a warning anyways and the user should be able to map it back themselves.